examples/async/index.js

```diff
@@ -1,15 +1,43 @@
 import 'babel-polyfill'
 import React from 'react'
-import { render } from 'react-dom'
+import ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
-import App from './containers/App'
 import configureStore from './store/configureStore'
 
 const store = configureStore()
+const rootEl = document.getElementById('root')
 
-render(
-  <Provider store={store}>
-    <App />
-  </Provider>,
-  document.getElementById('root')
-)
+let render = () => {
+  const App = require('./containers/App').default
```

Comment:
Also the reloading doesn't seem to bubble up quite as I expect - that is, if I have a component `Parent` that requires a component `Child`, and I have `module.hot.accept('components/Parent', ...`, it won't handle changes on `Child` (due to the `components/Parent` argument I guess?). Not really clear how all this is meant to work so a real world example in an app with multiple components, routing, redux etc. would be amazing :)